Coveralls doesn't work for booking-report module

Coveralls seems to not work with jacoco.xml generated by report-aggregate goal of org.jacoco:jacoco-maven-plugin. In particular, coveralls-maven-plugin:report goal cannot retrieve java class sources, although directories of sources are speciefied into <sourceDirectories> parameter. Error shown is:
`[ERROR] Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.3.0:report (report-coveralls) on project bookig-report: I/O operation failed: No source found for io/github/marcopaglio/booking/annotation/Generated.java`
When is runned on booking-aggregate the foolowing command:
`mvn clean install -Pcoverall -DrepoToken=<repoTokenHere>`

Utilities:

- github of jacoco-maven-plugin where are listed properties for configurations: [https://github.com/trautonen/coveralls-maven-plugin](url)
- open Issue of the same error (same github): [https://github.com/trautonen/coveralls-maven-plugin/issues/57](url)

--- a/booking-domain-module/pom.xml
+++ b/booking-domain-module/pom.xml
@@ -27,39 +27,18 @@
   </dependencies>
   
   <build>
-    <plugins>
-      
-    </plugins>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.jacoco</groupId>
+          <artifactId>jacoco-maven-plugin</artifactId>
+          <configuration>
+            <excludes>
+              <exclude>**/annotation/*</exclude>
+            </excludes>
+          </configuration>
+        </plugin>
+      </plugins>
+    </pluginManagement>
   </build>
-  
-    <profiles>
-      <!-- TO REMOVE AS SOON AS AGGREGATE COVERALLS WORKS -->
-      <profile>
-      <id>coveralls</id>
-      <build>
-        <plugins>
-          <plugin>
-            <groupId>org.jacoco</groupId>
-            <artifactId>jacoco-maven-plugin</artifactId>
-          </plugin>
-          <plugin>
-            <groupId>org.eluder.coveralls</groupId>
-            <artifactId>coveralls-maven-plugin</artifactId>
-            <configuration>
-              <dryRun>true</dryRun>
-            </configuration>
-            <executions>
-              <execution>
-                <id>report-coveralls</id>
-                <phase>verify</phase>
-                <goals>
-                  <goal>report</goal>
-                </goals>
-              </execution>
-            </executions>
-          </plugin>
-        </plugins>
-      </build>
-    </profile>
-  </profiles>
 </project>

--- a/booking-parent/pom.xml
+++ b/booking-parent/pom.xml
@@ -97,5 +97,23 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>coveralls</id>
+      <build>
+        <plugins>
+          <plugin>
+            <!-- configured in pluginManagement -->
+            <!-- JaCoCo report is required by coveralls-maven-plugin -->
+            <groupId>org.jacoco</groupId>
+            <artifactId>jacoco-maven-plugin</artifactId>
+          </plugin>
+          <plugin>
+            <!-- configured in pluginManagement -->
+            <groupId>org.eluder.coveralls</groupId>
+            <artifactId>coveralls-maven-plugin</artifactId>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
   </profiles>
 </project>

--- a/booking-report/pom.xml
+++ b/booking-report/pom.xml
@@ -8,7 +8,7 @@
   <artifactId>booking-report</artifactId>
   <packaging>pom</packaging>
   
-    <dependencies>
+  <dependencies>
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>booking-domain-module</artifactId>
@@ -23,6 +23,11 @@
           <plugin>
             <groupId>org.jacoco</groupId>
             <artifactId>jacoco-maven-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>**/annotation/*</exclude>
+              </excludes>
+            </configuration>
             <executions>
               <execution>
                 <id>report-aggregate</id>
@@ -45,6 +50,11 @@
             <!-- JaCoCo report is required by coveralls-maven-plugin -->
             <groupId>org.jacoco</groupId>
             <artifactId>jacoco-maven-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>**/annotation/*</exclude>
+              </excludes>
+            </configuration>
             <executions>
               <execution>
                 <id>report-aggregate</id>
@@ -72,9 +82,13 @@
               <jacocoReports>
                 <jacocoReport>${project.reporting.outputDirectory}/jacoco-aggregate/jacoco.xml</jacocoReport>
               </jacocoReports>
-              <scanForSources>true</scanForSources>
+              <!-- scanForSources doesn't find sources from ../booking-domain-module/ directory -->
               <sourceDirectories>
-                <sourceDirectory>../booking-domain-module/</sourceDirectory>
+                <!-- HERE FOR OTHER SOURCES -->
+                <!-- Path has to be in accord with source names in jacoco.xml file -->
+                <!-- Example: if name=io/github/marcopaglio/booking/model/Client -->
+                <!--          then: path is ../booking-domain-module/src/main/java/ -->
+                <sourceDirectory>../booking-domain-module/src/main/java/</sourceDirectory>
               </sourceDirectories>
             </configuration>
           </plugin>

